/**
 * mesher_cgal
 *
 * Copyright (C) 2013-  NUMA Engineering Ltd. (see AUTHORs file)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package mesherCGAL;

message CGALSettings {
    message Sphere {
        required float x = 1;
        required float y = 2;
        required float z = 3;
        required float r = 4;
    };

    message Zone {
        optional string file = 1;
        optional string name = 2;
        optional int32 index = 3;
        optional float characteristic_length = 4;
        optional float priority = 5;
        optional Sphere activity = 6;
        optional int32 inactivity_index = 7;
    };

    repeated float centre = 1;
    optional float centre_radius = 2;
    optional int32 tissue_id = 3;
    optional float bounding_radius = 4;
    optional bool tetrahedralize_only = 5;
    optional bool suppress_nef_loading = 6;
    optional bool boundary_tree = 7;
    optional bool omit_needle_tree = 8;
    optional bool omit_zone_tree = 9;
    optional bool dense_centre = 10;

    optional float near_field = 11;
    optional float far_field = 12;
    optional float zone_field = 13;
    optional float zone_radius = 33;
    optional float granularity = 14;

    optional string organ_file = 15;
    optional int32 organ_index = 28;
    optional string hepatic_artery = 16;
    repeated string vessels = 17;
    repeated int32 vessels_index = 31;
    repeated Zone zone = 35;
    repeated string zones = 18;
    repeated int32 zones_index = 32;
    optional string portal_vein = 19;

    optional string extent_file = 20;
    optional int32 extent_index = 29;
    repeated string needles = 21;
    repeated int32 needles_index = 30;

    optional string output_prefix = 22;
    optional bool output_vtk = 23;
    optional bool output_gmsh = 24;
    optional bool output_medit = 25;

    optional bool mark_zone_boundaries = 34;
    optional bool solid_zone = 38;

    optional bool number_from_zero = 26;

    optional bool verbose = 27;
    optional bool dump_settings = 36;

    optional float matching_tolerance = 37;
};
